modules/wpml/wpml-config.php

```diff
@@ -849,4 +826,136 @@ private function normalize_path( string $path ): string {
+		$metas_to_remove = array();
+
+		foreach ( $parsed_metas as $parsed_meta ) {
+			if ( 'copy' === $parsed_meta['action'] || ( ! $sync && in_array( $parsed_meta['action'], array( 'translate', 'copy-once' ), true ) ) ) {
+				$metas[] = $parsed_meta['name'];
+			} else {
+				$metas_to_remove[] = $parsed_meta['name'];
+			}
+		}
+
+		return array_diff( $metas, $metas_to_remove );
+	}
+
+	/**
+	 * Adds meta keys to export.
+	 *
+	 * @since 3.6
+	 *
+	 * @param array  $metas {
+	 *     An array containing meta keys to translate.
+	 *     Ex: array(
+	 *      'meta_to_translate_1' => 1,
+	 *      'meta_to_translate_2' => 1,
+	 *      'meta_to_translate_3' => array( ... ),
+	 *    )
+	 * }
+	 * @param string $xpath  Xpath to the meta fields in the xml files.
+	 * @return array
+	 *
+	 * @phpstan-param array<non-falsy-string, mixed> $metas
+	 * @phpstan-param non-falsy-string $xpath
+	 * @phpstan-return array<non-falsy-string, mixed>
+	 */
+	private function add_metas_to_export( array $metas, string $xpath ) {
+		$fields = $this->parse_xml_metas( $xpath );
+
+		foreach ( $fields as $field ) {
+			if ( 'translate' === $field['action'] ) {
+				$metas[ $field['name'] ] = 1;
+			}
+		}
+
+		return $metas;
+	}
+
+	/**
+	 * Adds encoding of metas.
+	 *
+	 * @since 3.6
+	 *
+	 * @param string[] $metas The list of encodings for each metas. Meta names are array keys, encodings are array values.
+	 * @param string   $xpath Xpath to the meta fields in the xml files.
+	 * @return string[]
+	 *
+	 * @phpstan-param array<non-falsy-string, non-falsy-string> $metas
+	 * @phpstan-param non-falsy-string $xpath
+	 */
+	private function add_metas_encodings( array $metas, string $xpath ): array {
+		$parsed_metas = $this->parse_xml_metas( $xpath );
+
+		foreach ( $parsed_metas as $parsed_meta ) {
+			if ( ! empty( $parsed_meta['encoding'] ) ) {
+				$metas[ $parsed_meta['name'] ] = $parsed_meta['encoding'];
+			}
+		}
+
+		return $metas;
+	}
+
+	/**
+	 * Parses all xml files for metas.
+	 * Results are cached for each `$xpath`.
+	 *
+	 * @since 3.6
+	 *
+	 * @param string $xpath Xpath to the meta fields in the xml files.
+	 * @return array
+	 *
+	 * @phpstan-param non-falsy-string $xpath
+	 * @phpstan-return list<ParsedMeta>
+	 */
+	private function parse_xml_metas( string $xpath ): array {
+		if ( isset( $this->parsed_metas[ $xpath ] ) ) {
+			return $this->parsed_metas[ $xpath ];
+		}
+
+		$this->parsed_metas[ $xpath ] = array();
+
+		foreach ( $this->xmls as $xml ) {
+			$custom_fields = $xml->xpath( $xpath );
+
+			if ( ! is_array( $custom_fields ) ) {
+				continue;
+			}
+
+			foreach ( $custom_fields as $custom_field ) {
+				$name = (string) $custom_field;
+
+				if ( empty( $name ) ) {
+					continue;
+				}
+
+				$data = array(
+					'name'     => $name,
+					'action'   => $this->get_field_attribute( $custom_field, 'action' ),
+					'encoding' => $this->get_field_attribute( $custom_field, 'encoding' ),
+				);
```

Comment:
Wouldn't it be easier to manipulate?
```suggestion
				$data[ $name ] = array(
					'action'   => $this->get_field_attribute( $custom_field, 'action' ),
					'encoding' => $this->get_field_attribute( $custom_field, 'encoding' ),
				);
```

Reply:
Why not.

Reply:
Done in [43c8b85](https://github.com/polylang/polylang/pull/1402/commits/43c8b851293473044f894f9d36e115e3cf7669fd).